resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverBuilder.java

```diff
@@ -33,24 +33,24 @@
 /**
  * A {@link DnsNameResolver} builder.
  */
-public final class DnsNameResolverBuilder {
-
-    private final EventLoop eventLoop;
-    private ChannelFactory<? extends DatagramChannel> channelFactory;
-    private InetSocketAddress localAddress = DnsNameResolver.ANY_LOCAL_ADDR;
-    private DnsServerAddresses nameServerAddresses;
-    private DnsCache resolveCache;
-    private Integer minTtl;
-    private Integer maxTtl;
-    private Integer negativeTtl;
-    private long queryTimeoutMillis = 5000;
-    private InternetProtocolFamily[] resolvedAddressTypes = DnsNameResolver.DEFAULT_RESOLVE_ADDRESS_TYPES;
-    private boolean recursionDesired = true;
-    private int maxQueriesPerResolve = 3;
-    private boolean traceEnabled;
-    private int maxPayloadSize = 4096;
-    private boolean optResourceEnabled = true;
-    private HostsFileEntriesResolver hostsFileEntriesResolver = HostsFileEntriesResolver.DEFAULT;
+public abstract class DnsNameResolverBuilder<T extends DnsNameResolverBuilder<T>> {
```

Comment:
Could we rename the builder classes to `AbstractDnsNameResolverBuilder` and `DnsNameResolverBuilder`, like we did in `AbstractHttp2ConnectionHandlerBuilder` and `Http2ConnectionHandlerBuilder`?